src/pymor/reductors/loewner.py

```diff
@@ -0,0 +1,309 @@
+        elif ordering == 'regular':
+            idx = np.arange(s.shape[0])
+
+        if partitioning == 'even-odd':
+            return (idx[::2], idx[1::2])
+        elif partitioning == 'half-half':
+            idx_split = np.array_split(idx, 2)
+            return (idx_split[0], idx_split[1])
+
+
+def loewner_quadruple(s, Hs, partitioning='even-odd', ordering='regular', conjugate=True, ltd=None, rtd=None):
+    r"""Constructs a Loewner quadruple as |NumPy arrays|.
+
+    The Loewner quadruple :cite:`ALI17`
+
+    .. math::
+        (\mathbb{L},\mathbb{L}_s,V,W)
+
+    consists of the Loewner matrix :math:`\mathbb{L}`, the shifted Loewner matrix
+    :math:`\mathbb{L}_s`, left interpolation data :math:`V` and right interpolation data :math:`W`.
+
+    Parameters
+    ----------
+    s
+        |Numpy Array| of shape (n,) containing the frequencies.
+    Hs
+        |Numpy Array| of shape (n, p, m) for MIMO systems with p outputs and m inputs or
+        |Numpy Array| of shape (n,) for SISO systems where the |Numpy Arrays| resemble the transfer
+        function samples. Alternatively, |TransferFunction| or `Model` with `transfer_function`
+        attribute.
+    partitioning
+        `str` or `tuple` of length 2. Strings can either be 'even-odd' or 'half-half' defining
+        the splitting rule. A user-defined partitioning can be defined by passing a tuple of the
+        left and right indices. Defaults to `even-odd`.
+    ordering
+        The ordering with respect to which the splitting rule is executed. Can be either
+        'magnitude', 'random' or 'regular'. Defaults to 'regular'.
+    conjugate
+        Whether to guarantee realness of reduced |LTIModel| by keeping complex conjugates in the
+        same partitioning or not. If `True` will automatically generate conjugate data if necessary.
+    ltd
+        |Numpy Array| representing left tangential directions. If `None` tangential directions will
+        be chosen randomly (normal distribution).
+    rtd
+        |Numpy Array| representing right tangential directions. If `None` tangential directions will
+        be chosen randomly (normal distribution).
+
+    Returns
+    -------
+    L
+        Loewner matrix.
+    Ls
+        Shifted Loewner matrix.
+    V
+        Left interpolation data.
+    W
+        Right interpolation data.
+    """
+    assert isinstance(s, np.ndarray)
+    if hasattr(Hs, 'transfer_function'):
+        Hs = Hs.transfer_function
+    assert isinstance(Hs, TransferFunction) or isinstance(Hs, np.ndarray) or isinstance(Hs, list)
+
+    assert partitioning in ('even-odd', 'half-half') or len(partitioning) == 2
+    assert ordering in ('magnitude', 'random', 'regular')
+
+    if isinstance(Hs, TransferFunction):
+        Hss = np.empty((len(s), Hs.dim_output, Hs.dim_input), dtype=s[0].dtype)
+        for i, ss in enumerate(s):
+            Hss[i] = Hs.eval_tf(ss)
+        Hs = Hss
+    else:
+        Hs = np.atleast_3d(Hs)
```

Comment:
Have you tested if this works well? I remember having trouble with `atleast_3d` because it adds a dimension to both sides, i.e. an the shape of an array with initial shape `(n,)` becomes `(1, n, 1)` if remember correctly.